crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -1537,6 +1544,14 @@ struct SolveState {
     /// This keeps track of the set of versions for each package that we've
     /// already visited during resolution. This avoids doing redundant work.
     added_dependencies: FxHashMap<PubGrubPackage, FxHashSet<Version>>,
+    /// The marker expression that created this state.
+    ///
+    /// The root state always corresponds to a marker expression that is
+    /// always `true` for every `MarkerEnvironment`.
+    ///
+    /// In non-universal mode, forking never occurs and so this marker
+    /// expression is always `true`.
```

Comment:
OK, I added a paragraph. Hopefully it makes its use a little clearer:

```rust
    /// The marker expression that created this state.
    ///
    /// The root state always corresponds to a marker expression that is always
    /// `true` for every `MarkerEnvironment`.
    ///
    /// In non-universal mode, forking never occurs and so this marker
    /// expression is always `true`.
    ///
    /// Whenever dependencies are fetched, all requirement specifications
    /// are checked for disjointness with the marker expression of the fork
    /// in which those dependencies were fetched. If a requirement has a
    /// completely disjoint marker expression (i.e., it can never be true given
    /// that the marker expression that provoked the fork is true), then that
    /// dependency is completely ignored.
```